### PR TITLE
Moved CC macros to public API

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Moved CC macros for transformations to public API (to reuse their in plugin transformations)

### CODE REVIEW

File: inference-engine/src/transformations/src/itt.hpp [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 #include <openvino/cc/selective_build.h>
 #include <openvino/itt.hpp>
+#include <ngraph/pass/itt.hpp>
```
Comment (ilya-lavrenov) [inline inference-engine/src/transformations/src/itt.hpp:14]: Need to replace code in snippets then

**Review by nkogteva — APPROVED** [review]

File: inference-engine/src/transformations/src/itt.hpp (hunk 1 above)
Comment (ilyachur) [reply inference-engine/src/transformations/src/itt.hpp:14]: We didn't have CC macros in snippets. I added these macros to template plugin.

Comment (ilya-lavrenov) [reply inference-engine/src/transformations/src/itt.hpp:14]: Really?
https://github.com/openvinotoolkit/openvino/blob/315c8d4eec75d1efbce553434ed5db5df6d97665/inference-engine/src/snippets/src/itt.hpp#L33-L59

File: ngraph/core/include/ngraph/pass/itt.hpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/cc/selective_build.h>
+#include <openvino/itt.hpp>
```
Comment (ilya-lavrenov) [inline ngraph/core/include/ngraph/pass/itt.hpp:8]: these headers are NOT a part of release package

File: inference-engine/src/transformations/src/itt.hpp (hunk 1 above)
Comment (ilyachur) [reply inference-engine/src/transformations/src/itt.hpp:14]: Good catch

File: ngraph/core/include/ngraph/pass/itt.hpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/pass/itt.hpp:8]: Yeah right. I will think how to enable this file only for open source.

Comment (ilya-lavrenov) [reply ngraph/core/include/ngraph/pass/itt.hpp:8]: looks like we need to move all this stuff to `openvino/itt`

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**